src/providers/create-new/fields/openstack/credentials/ApplicationCredentialIdCredentialsFields.tsx

```diff
@@ -0,0 +1,116 @@
+
+const ApplicationCredentialIdCredentialsFields: FC = () => {
+  const { t } = useForkliftTranslation();
+  const { control } = useCreateProviderFormContext();
+  const [secretHidden, setSecretHidden] = useState<boolean>(true);
+
+  const {
+    field: { onChange: onAppCredIdChange, value: appCredIdValue },
+    fieldState: { error: appCredIdError },
+  } = useController({
+    control,
+    name: ProviderFormFieldId.OpenstackApplicationCredentialId,
+    rules: {
+      required: t('Application credential ID is required'),
+      validate: (val: string | boolean | undefined) =>
+        validateOpenstackField(OpenstackSecretFieldsId.ApplicationCredentialId)(val),
+    },
+  });
+
+  const {
+    field: { onChange: onSecretChange, value: secretValue },
+    fieldState: { error: secretError },
+  } = useController({
+    control,
+    name: ProviderFormFieldId.OpenstackApplicationCredentialSecret,
+    rules: {
+      required: t('Application credential secret is required'),
+      validate: (val: string | boolean | undefined) =>
+        validateOpenstackField(OpenstackSecretFieldsId.ApplicationCredentialSecret)(val),
+    },
+  });
+
+  return (
+    <Form>
+      <FormGroupWithHelpText
+        label={t('Application credential ID')}
+        isRequired
+        fieldId={ProviderFormFieldId.OpenstackApplicationCredentialId}
+        validated={getInputValidated(appCredIdError)}
+        helperTextInvalid={appCredIdError?.message}
+      >
+        <TextInput
+          type="text"
+          id={ProviderFormFieldId.OpenstackApplicationCredentialId}
+          value={appCredIdValue ?? ''}
+          onChange={(_event, val) => {
+            onAppCredIdChange(val);
+          }}
+          validated={getInputValidated(appCredIdError)}
+          data-testid="openstack-app-cred-id-input"
+        />
+      </FormGroupWithHelpText>
+
+      <FormGroupWithHelpText
+        label={t('Application credential secret')}
+        isRequired
+        fieldId={ProviderFormFieldId.OpenstackApplicationCredentialSecret}
+        validated={getInputValidated(secretError)}
+        helperTextInvalid={secretError?.message}
+      >
+        <InputGroup>
+          <TextInput
+            type={secretHidden ? 'password' : 'text'}
+            id={ProviderFormFieldId.OpenstackApplicationCredentialSecret}
+            value={secretValue ?? ''}
+            onChange={(_event, val) => {
+              onSecretChange(val);
+            }}
+            validated={getInputValidated(secretError)}
+            data-testid="openstack-app-cred-secret-input"
+          />
+          <Button
+            variant={ButtonVariant.control}
+            onClick={() => {
+              setSecretHidden(!secretHidden);
+            }}
+            aria-label={secretHidden ? 'Show secret' : 'Hide secret'}
+          >
+            {secretHidden ? <EyeIcon /> : <EyeSlashIcon />}
+          </Button>
+        </InputGroup>
+      </FormGroupWithHelpText>
```

Comment:
extract to a component since this pattern is re-used accross different components and name ot OpenstackSecretField